docs/architecture/adr-005-modular-server.md

```diff
@@ -0,0 +1,169 @@
+# ADR 005: Modular Server
+
+## Changelog
+
+- 2020/12/22: Initial Draft
+
+## Author
+
+- Federico Kunze ([@fedekunze](https://github.com/fedekunze))
+
+## Status
+
+Draft, Not Implemented
+
+## Abstract
+
+This document outlines a modular and standard approach for using servers on the SDK, so that
+applications can also run custom services besides the default ones provided by the SDK.
+
+## Context
+
+The current `Server` implementation allows for applications to register routes, API and gRPC
+services to the SDK. The problem with the current approach is that the SDK uses a concrete type,
+which prevents other SDK-based blockchains to extend the services provided by the application
+without forking the application `start` command (eg: `<app>d start`).
+
+This modularity and extensibility relies on the following limitations:
+
+    - Consistent services
+    - App configuration extensibility
+
+### Consistent services
```

Comment:
How about renaming to: _Service setup_? Or _Service Registration Process_,   _Consisten Service Registration_...